test/core/image_processing/box_filter.cpp

```diff
@@ -0,0 +1,67 @@
+//
+// Copyright 2019 Miral Shah <miralshah2211@gmail.com>
+//
+// Use, modification and distribution are subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+//
+
+#define BOOST_TEST_MODULE test_image_processing_box_filter
+#include "unit_test.hpp"
+
+#include <boost/gil/image_view.hpp>
+#include <boost/gil/algorithm.hpp>
+#include <boost/gil/gray.hpp>
+#include <boost/core/lightweight_test.hpp>
```

Comment:
@miralshah365 Since Boost.Test is used, `#include <boost/core/lightweight_test.hpp>` must be removed, otherwise `BOOST_TEST` and other macros will clash.